_posts/2024-07-16-17.0.0-release.md

```diff
@@ -0,0 +1,222 @@
+- Non-CPU data with offsets, such as list and binary data, can now be properly
+  sent over IPC (GH-42198).
+
+### IPC
+
+- Flatbuffers serialization is now more deterministic (GH-40361).
+
+### Parquet
+
+- A crash was fixed when reading an invalid Parquet file where columns claim to
+  be of different lengths (GH-41317).
+- Definition and repetition levels are now more strictly checked, avoiding later
+  crashes when reading an invalid Parquet file (GH-41321).
+- A crash was fixed when reading an invalid encrypted Parquet file (GH-43070).
+- Fixed a bug where `DeltaLengthByteArrayEncoder::EstimatedDataEncodedSize` could
+  return an invalid estimate in some situations (GH-41545).
+- Delimiting records is now faster for columns with nested repeating (GH-41361).
+
+### Substrait
+
+- Support for more Arrow data types was added: some temporal types, half floats,
+  large string and large binary (GH-40695).
+
+## C# notes
+
+- The performance of building Decimal arrays using SqlDecimal values was improved for .NET 7+ (GH-41349)
+- Scalar arrays now implement `ICollection<T?>` (GH-38692)
+- Concatenating arrays with a non-zero offset with ArrowArrayConcatenator was fixed (GH-41164)
+- Concatenating union arrays with ArrowArrayConcatenator was fixed (GH-41198)
+- Accessing values of decimal arrays with a non-zero offset was fixed (GH-41199)
+
+## Go Notes
+
+### Bug Fixes
+
+#### Arrow
+
+- Prevent exposure of invalid Go pointers in CGO code ([GH-43062](https://github.com/apache/arrow/issues/43062))
+- Fix memory leak for 0-length C array imports ([GH=41534](https://github.com/apache/arrow/issues/41534))
+- Ensure statement handle is updated so stateless prepared statements work properly ([GH-41427](https://github.com/apache/arrow/issues/41427))
+
+#### Parquet
+
+- Fix memory leak in BufferedPageWriter ([GH-41697](https://github.com/apache/arrow/issues/41697))
+- Fix performance regression in PooledBufferWriter ([GH-41541](https://github.com/apache/arrow/issues/41541))
+
+### Enhancements
+
+#### Arrow
+
+- Arrow Schemas and Records can now be created from Protobuf messages ([GH-40494](https://github.com/apache/arrow/issues/40494))
+
+#### Parquet
+
+- Performance improvement for BitWriter VlqInt ([GH-41160](https://github.com/apache/arrow/pull/41160))
+
+## Java notes
+
+**Some changes are coming up in the next version, Arrow 18. Java 8 support will be removed. The version of the flight-core artifact with shaded gRPC will no longer be distributed.**
+
+- Basic support for ListView (GH-41287) and StringView (GH-40339) has been added. These types should still be considered experimental. 
+
+## JavaScript notes
+
+- General maintenance. Clean up packaging ([GH-39722](https://github.com/apache/arrow/issues/39722)), update dependencies ([GH-41905](https://github.com/apache/arrow/issues/41905)).
+
+## Python notes
+
+- Added support for Numpy 2.0 (GH-42170, GH-41924).
+- Added support for Emscripten via Pyodide (GH-41910).
```

Comment:
We usually have expanded Python notes, are we happy to publish with those?
The rest of the blog post looks good to me

Reply:
Oh wow, you're right. Let me update this.

Reply:
```suggestion
## Python notes

Compatibility notes:
* To ensure Python 3.13 compatibility, _Py_IsFinalizing has been replaced with a public API (GH-41475).
* The C Data Interface now supports CUDA devices (GH-40384).

New features:
* Added support for Emscripten via Pyodide (GH-41910).

Other improvements:
* The ParquetWriter added the store_decimal_as_integer option (GH-42168).
* The Float16 logical type is supported in Parquet (GH-42016).
* Exposed bit_width and byte_width to extension types (GH-41389).
* Added bindings for Device and MemoryManager classes (GH-41126).
* The PyCapsule interface now exposes the device interface (GH-38325).
* Various PyArrow APIs have been updated to work with non-CPU architectures gracefully. (GH-42112, GH-41664, GH-41662, 

Relevant bug fixes:
* Fixed Numpy 2.0 compatibility issues (GH-42170, GH-41924).
* Fixed sporadic as_of join failures (GH-41149).
* Fixed a bug in RecordBatch.filter() when passing a ChunkedArray, which would cause a segfault (GH-38770).
* Fixed a bug in RecordBatch.from_arrays() when passing a storage array, which would cause a segfault (GH-37669).
* Fixed a bug where constructing a MapArray from Array could drop nulls (GH-41684).
* FIxed a bug where RunEndEncodedArray.from_arrays fails if run_ends are pyarrow.Array (GH-40560).
* FIxed a regression introduced in PyArrow v16 in RecordBatchReader.cast() (GH-41884).
```